### PR TITLE
Ma732 driver

### PR DESCRIPTION
**Pull request by Dayren43**
# MA732 Driver
This pull request adds in a driver supporting the MA732 hall effect angle sensor.

# I implemented the driver to enable usage of the MA732

The sensor is needed for angle measurements. The driver supports the most common commands
- read angle
- write register
- read register

As well as some niceties 
- set zero angle
- reset registers

## Testing
Not yet tested on hardware

## TODOs
Test the examples to validate it's still functional on hardware
- [ ] CAN
- [x] MA732

### CODE REVIEW

File: examples/MA732.rs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            pins,
+            Frequency::K250,
+            embedded_hal::spi::MODE_0,
+        );
+
+        let driver = Driver::new(cs);
+
+        worker::spawn().unwrap();
+
+        //blast::spawn().ok().unwrap();
```
Comment (ivajon) [inline examples/MA732.rs]: Remove this

File: examples/MA732.rs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+        (
+            Shared {
+                // Initialization of shared resources go here
+            },
+            Local {
+                spim,
+                driver, //cs
```
Comment (ivajon) [inline examples/MA732.rs]: Remove cs comment

File: src/drivers/MA732.rs [hunk 3]
```diff
@@ -0,0 +1,205 @@
+#![allow(clippy::future_not_send)]
+use core::f32;
+
+use embedded_hal::{digital::OutputPin, spi::SpiBus};
+use rtic_monotonics::{fugit::ExtU64, Monotonic};
+
+///Struct containing the needed pins to drive MA732 sensor.
+pub struct Driver</* SPI: SpiBus, */ PIN: OutputPin> {
```
Comment (ivajon) [inline src/drivers/MA732.rs]: Remove /*SPI: SpiBus, */

File: src/drivers/MA732.rs [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    /// Spi configured with MISO, MOSI and clk pins.
+    // pub spi: SPI,
```
Comment (ivajon) [inline src/drivers/MA732.rs:10]: Remove spi field

File: src/drivers/MA732.rs [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    /// maybe be passed to the read/write commands to make it able to point to
+    /// multiple senores.
+    pub const fn new(cs: PIN) -> Self {
+        Self { cs }
+    }
+
+    /// Reads the angle from the MA732 sensor.
+    ///
+    /// returns the angle in radians.
+    /// ### note
```
Comment (ivajon) [inline src/drivers/MA732.rs:72]: Add new lines around the rubric

```rust
    /// returns the angle in radians.
    ///
    /// ### Note
    ///
```

File: src/drivers/MA732.rs [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+        .await;
+        self.write_register::<T, SPI, NOM, DENOM>(
+            Register::RotationDIrection,
+            spi,
+            ROTATION_DIRECTION,
+        )
+        .await;
+    }
+}
+#[allow(clippy::cast_sign_loss)]
```
Comment (ivajon) [inline src/drivers/MA732.rs:199]: Add new line above this to be consistent.

File: src/drivers/MA732.rs [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+#[allow(clippy::cast_possible_truncation)]
+const fn radians_to_u16(radians: f32) -> u16 {
+    let radians = radians % f32::consts::TAU;
+    ((radians / f32::consts::TAU) * u16::MAX as f32) as u16
+}
+#[allow(clippy::cast_possible_truncation)]
```
Comment (ivajon) [inline src/drivers/MA732.rs:206]: ditto

**Review by ivajon — CHANGES_REQUESTED** [review]
Looks good in general, just a few nits.

**Comment by ivajon** [comment]
Merge plan:

- [x] Fix the nits
- [x] Test driver to see that it reports correct values.
- [ ] Test the can driver again with this branch to ensure nothing broke. Or if @wiiggee1 thinks it's fine we simply ignore this test.

**Review by ivajon — APPROVED** [review]

**Review by wiiggee1 — APPROVED** [review]
Would mark this as OK for now. If you made some "unsafe proc macro changes". We might push that to another branch.

**Pull request merged**